### PR TITLE
[T-2] task 2

### PR DESCRIPTION
**Pull request by radostnomne**
s3 - http://konyaev-site.s3-website-eu-west-1.amazonaws.com/
cloudfront - https://da191rapabbpu.cloudfront.net/

command to deploy:
`npm run cloudfront:update:build:deploy`

### CODE REVIEW

**Comment by KarneyenkaDzmitry** [comment]
Hello Pavel,
 I will appreciate it if you also share the links to the first (manual) created s3 and CloudFront.

**Comment by radostnomne** [comment]
@KarneyenkaDzmitry  sure
s3 - http://manual-task-two.s3-website-eu-west-1.amazonaws.com
cloudfront - d1zg3fljof5ysm.cloudfront.net

File: serverless-single-page-app-plugin/index.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  runAwsCommand(args) {
+    let command = 'aws';
+    if (this.serverless.variables.service.provider.region) {
+      command = `${command} --region ${this.serverless.variables.service.provider.region}`;
+    }
+    if (this.serverless.variables.service.provider.profile) {
+      command = `${command} --profile ${this.serverless.variables.service.provider.profile}`;
+    }
+    const result = spawnSync(command, args);
```
Comment (KarneyenkaDzmitry) [inline serverless-single-page-app-plugin/index.js:47]: You can also use the following approach:
## Known Issues

1. `Type Error ---------------------------------------------`</br>
   `TypeError: Cannot read property 'toString' of null` `at ServerlessPlugin.runAwsCommand (C:\Users\Dzmitry_Karneyenka\OneDrive - EPAM\BackUp\AWS and Nodejs\nodejs-aws-fe\serverless-single-page-app-plugin\index.js:48:34)` `at ServerlessPlugin.invalidateCache (C:\Users\Dzmitry_Karneyenka\OneDrive - EPAM\BackUp\AWS and Nodejs\nodejs-aws-fe\serverless-single-page-app-plugin\index.js:135:30)`</br>.
   `For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.` </br>
   Solutions: In serverless-single-page-app-plugin/index.js add shell:true option when spawnSync.</br>
   from: `const result = spawnSync(command, args);`</br>
   to: `const result = spawnSync(command, args, {shell: true});`</br>
   [Nodejs Documentation](https://nodejs.org/docs/latest-v12.x/api/child_process.html#child_process_child_process_spawnsync_command_args_options);

**Comment by KarneyenkaDzmitry** [comment]
@radostnomne , Well done.

**Comment by ArtiFisher** [comment]
Haven't found any issues in this PR. Good job!

**Pull request merged**